### PR TITLE
Boiler plate for mentored projects documentation

### PR DESCRIPTION
**Pull request by MridulS**

### CODE REVIEW

File: doc/developer/projects.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+==================
+
+This page maintains a list of mentored project ideas that contributors can work
+on if they are interested in contributing to the NetworkX project. Feel free to
+suggest any other idea if you are interested on the
+`NetworkX GitHub discussions page <https://github.com/networkx/networkx/discussions>`__ 
+
+These ideas can be used as projects for Google Summer of Code, Outreachy,
+NumFOCUS Small Development Grants and university course/project credits (if
+your university allows contribution to open source for credit).
```
Comment (rossbar) [inline doc/developer/projects.rst:11]: A bit more background here would be good: something about how these projects are expected to be larger in scope than a single PR, i.e. appropriate for: <your list, GSoc, semester projects, etc.>.

**Review by rossbar — APPROVED** [review]
I am +1 on adding something about mentored projects directly to the developer guide. I think it helps advertise the ideas and demonstrates willingness to help others who are interested in contributing.

I took the liberty of cleaning up some links+adding linebreaks and I'm sure there's some other polishing to be done, but I like the idea of having this in the docs.

**Comment by stefanv** [comment]
Thanks @MridulS, I like this page and hope to see it grow over time.  @rossbar linking this to the contributor guide is a good idea.

**Comment by rossbar** [comment]
> linking this to the contributor guide is a good idea.

Agreed - when this goes in I'll add a bullet to the FAQ linking to it.

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**